Source/ButtonMerchant.swift

```diff
@@ -72,7 +72,12 @@ final public class ButtonMerchant: NSObject {
 
      */
     @objc public static func configure(applicationId: String) {
-        core.applicationId = applicationId
+        guard let validAppId = ApplicationId(applicationId) else {
+            let error = ConfigurationError.invalidApplicationId(appicationId: applicationId)
+            print("⚠️ Button :: \(error.localizedDescription)")
+            return
```

Comment:
You should **not** return. We shouldn't be preempting configuration without having a completion callback in place.

Reply:
You have to return from a guard, and there is nothing to do without an appId.

Reply:
it's the same thing as

```
if let appId = ApplicationId(applicationId) {
   core.applicationId = appId
}

```

Reply:
And, it's critical we don't store the garbage appId or zero network requests will work.

Reply:
So it looks like we have a divergence [at this point](https://github.com/button/button-merchant-android/pull/52#discussion_r461937545) on whether or not we affect configuration—which includes only the setting of the appId internally. 

If we received an appId here of `my app id`, for example, we would be unable to fulfill a request to the network due to an invalid base url. However, if we choose to _not set the appId_, we'll use the base url _without_ the appId subdomain which will succeed and make it easier to identify the error by receiving requests with a missing appId.

Reply:
@wessmith But the `application_id` is required anyway. Not necessarily on the subdomain but in the request body. So we'd need to prevent configuration if an invalid app ID is provided.

Reply:
Allowing the request to be made, in the case of the deep link event, would be much easier to identify the error than silent failure in the wild.